core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -447,8 +451,10 @@ protected Location onReplicaShard(BulkShardRequest request, IndexShard indexShar
             if (item.request() instanceof IndexRequest) {
                 IndexRequest indexRequest = (IndexRequest) item.request();
                 try {
-                    Engine.Index operation = TransportIndexAction.executeIndexRequestOnReplica(indexRequest, indexShard);
-                    location = locationToSync(location, operation.getTranslogLocation());
+                    WriteResult<IndexResponse> result = TransportIndexAction.executeIndexRequestOnReplica(indexRequest, indexShard);
+                    if (result.success()) {
+                        location = locationToSync(location, result.getLocation());
+                    }
                 } catch (Exception e) {
```

Comment:
can you double check we need to protect against ignoreReplicaExceptions? I think we do that high up the stack.